pytorch_lightning/accelerators/accelerator.py

```diff
@@ -348,7 +348,12 @@ def connect_precision_plugin(self, plugin: PrecisionPlugin) -> None:
 
     def to_device(self, batch: Any) -> Any:
         """Pushes the batch to the root device"""
-        return self.batch_to_device(batch, self.root_device)
+        # Todo (tchaton) Better fix
+        is_dict = isinstance(batch, dict)
```

Comment:
Could you add a comment for this? What issues were you seeing before?

Reply:
In Flash, for HF transformers batch (dictionaries with "inputs_ids", "labels", ..), I was receiving only the keys after to_device.

Reply:
could you add a test case for this to ensure the move works as expected?